File: java/src/main/java/Set.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;

public class  Set {

    private List<String> elements;

    private Set() {
        this.elements = new ArrayList<>();
    }

    public Set(List<String> elements){
        this.elements = new ArrayList<>(elements);
    }

    public List<String> getElements() {
        return elements;
    }


    @Override
    public String toString() {
        return elements.toString();
    }
    public boolean ehVazio(){
        return elements.isEmpty();
    }

    public Set uniaoConj(Set b){
        List<String> union = new ArrayList<>(elements);
        union.addAll(b.elements.stream().filter(x -> !elements.contains(x)).toList());
        return new Set(union);
    }

    public Set intersConj(Set b){
        List<String> inters = new ArrayList<>(elements);
        inters.retainAll(b.elements);
        return new Set(inters);
    }

    public Set diffConj(Set b){
        List<String> diff = new ArrayList<>(elements);
        diff.removeAll(b.elements);
        return new Set(diff);
    }

    public Set diffSimConj(Set b){
        List<String> diffSim = new ArrayList<>(diffConj(b).elements);
        diffSim.addAll(b.diffConj(this).elements);
        return new Set(diffSim);
    }

    public void addConj(String x){
        if(!elements.contains(x))
            elements.add(x);
    }

    public void remConj( String x){
        if(elements.contains(x))
            elements.remove(x);
    }

    public Integer tamConj(){
        return elements.size();
    }

    public boolean ehElemConj(String x){
        return elements.contains(x);
    }
}

File: python/Set.py
# ehVazio(A)

def ehVazio(A):
    return not bool(A)

# uniaoConj(A,B)
def uniaoConj(A, B):
    for i in B:
        if i not in A:
            A.append(i)
    return A

# intersConj retorna A ∩ B.
def intersConj(A,B): 
    C = []
    for i in B:
        if i in A:
            C.append(i)
    return C

def diffConj(A,B):
    C = []
    for i in A:
        if i not in B:
            C.append(i)
    return C

def diffSimConj(A,B):
    diffA = diffConj(A,B)
    diffB = diffConj(B,A)
    return uniaoConj(diffA,diffB)

def addConj(A,x):
    if x not in A:
        A.append(x)
    return A

def remConj(A,x):
    if x in A:
        A.remove(x)
    return A

def tamConj(A):
    return len(A)

def ehElemConj(A,x):
    return x in A